### PR TITLE
Refactored rendered rolls for Astral and small fixes around.

### LINKED ISSUE #538: Beyond20-AstralTT Roll Calculation Issues
@adriangaro 

Long time Beyond20 user, first time poster! Loving the v2.2,

I am finding a few small issues, detailed below.

1) the dice roll calculation(?) in astral seems all whack, (img attached)
2) Clicking on a non-roll oriented spell (shield, for example) does not send anything to Astral.
3) The "Click to roll damage" box in astral doesn't seem to work off the B20 roll - it just opens recurring chat windows. 
4) When making a skill check, it has a line of text in the description box that just says "name         roll" 
5) Rolling damage, on damage oriented spells sends the spell description to astral, but not the damage dice roll.

I think thats all for now, please let me know if there's anything I can do to help - love Beyond 20!

Screen Shot 2020-10-04 at 1 05 11 pm

### PR DESCRIPTION
**Pull request by adriangaro**
This PR should be targeted to fix #538 and some other issues/improvements I wanted to fix/implement.

This would also fix the issue reported in the astral chat which caused spell cards to not be able to roll damage when using rendered rolls.

### CODE REVIEW

**Comment by adriangaro** [comment]
So an update with the latest changes in the PR, improved error handling here and there.

Resolved an issue notified by @Alkalithe in discord. His issue stemmed  from the possibility of having multiple entries in the firebase indexedb which didn't store the user auth token.

Changed rendered rolls to set the roll result similar to the way @Redmega recommended in #538.

Rendered rolls now should add the required roll info to the damage button, as the damage button would not be working after refresh/would redirect the user to an non-existent url. This is achieved by trimming the data stored and compressing it in the anchor href. The compression is mandatory as we are still bound by the 2048 character limit.

Some improvement on acquiring the iframe content document. Sometimes the initial reference to the iframe was refreshed and caused some error down the line.

File: src/astral/page-script.js [hunk 1]
```diff
@@ -93,21 +119,32 @@ async function handleRenderedRoll(request) {
...
-    await postChatMessage({characterName: request.character, message, ...getDecoration(request.request.type), title, whisper: request.whisper == WhisperType.YES });
-    if (rollDamages) {
-        await hookRollDamages(rollDamages, request);
+
+    
+    if (!originalRequest.rollDamage && request.open && request.description) {
+        message = `${message}
+
+
+${parseDescription(request, request.description)}`
```
Comment (kakaroto) [inline src/astral/page-script.js:144]: Indentation issue here

**Review by kakaroto — APPROVED** [review]

**Comment by kakaroto** [comment]
Thanks for the changes. I had a very quick look, not much for me to comment on, you're the maintainer for Astral and it generally looked good enough, so I'll merge it, thanks! 👍 
I'd appreciate if you could list here as a response comment the list of features and bugfixes that went into this PR in the same sort of format I usually do in the changelog, so that I can copy it into the changelog on release date.

Thanks!

**Comment by adriangaro** [comment]
Change log for Astral:

- **Bugfix**: Fixed digital dice rolls / rendered rolls for Astral. Reverted to normal behaviour when no dice are rolled.
- **Bugfix**: Fixed Astral access token acquisition in some specific edge cases.
- **Bugfix**: Fixed some issues related to slow loading of chat widget in Astral.
- **Bugfix**: Changed rendered roll formulas in Astral using the syntax recommended by @Redmega.
- **Bugfix**: Fixed some errors related to speak as character functionality.
- **Feature/Bugfix**: Changed implementation of roll buttons (when using digital dice) to work after reloading the page. They should work consistently now.

**Pull request merged**